src/smartcontracts/typesystem/types.ts

```diff
@@ -10,18 +10,20 @@ export class Type {
 
     private readonly name: string;
     private readonly typeParameters: Type[];
-    protected readonly cardinality: TypeCardinality;
+    protected readonly metadata: any;
```

Comment:
maybe move the  protected member last so that they are grouped by access modifiers?

Reply:
also, as discussed, the type can be `string`.

Reply:
done